src/bp/core/services/nlu/intent-service.ts

```diff
@@ -18,7 +19,45 @@ export class IntentService {
 
   public async getIntents(botId: string): Promise<sdk.NLU.IntentDefinition[]> {
     const intentNames = await this.ghostService.forBot(botId).directoryListing(INTENTS_DIR, '*.json')
-    return Promise.mapSeries(intentNames, n => this.getIntent(botId, n))
+    const intentsFromFiles = await Promise.mapSeries(intentNames, n => this.getIntent(botId, n))
+    const intentsFromFlows = await this.getIntentsFromFlows(botId)
+    return [...intentsFromFiles, ...intentsFromFlows]
+  }
+
+  private async getIntentsFromFlows(botId: string): Promise<sdk.NLU.IntentDefinition[]> {
+    const flowsPaths = await this.ghostService.forBot(botId).directoryListing(FLOWS_DIR, '*.flow.json')
+    const flows: sdk.Flow[] = await Promise.map(flowsPaths, async (flowPath: string) => ({
+      // @ts-ignore
+      name: flowPath.replace(/.flow.json$/i, ''),
+      ...(await this.ghostService.forBot(botId).readFileAsObject<FlowView>(FLOWS_DIR, flowPath))
+    }))
+
+    const intentsByName: Dic<sdk.NLU.IntentDefinition> = {}
+
+    for (const flow of flows) {
+      const topicName = flow.name.split('/')[0]
+
+      for (const node of flow.nodes.filter(x => x.type === 'trigger')) {
+        const tn = node as sdk.TriggerNode
+        const conditions = tn?.conditions.filter(x => x?.id === 'user_intent_is')
+
+        for (let i = 0; i < conditions.length; i++) {
+          const intentName = `${topicName}/${flow.name}/${tn?.name}/${i}`
+          if (intentsByName[intentName]) {
+            throw new Error(`Duplicated intent with name "${intentName}"`)
+          }
+          intentsByName[intentName] = {
+            contexts: [topicName],
+            filename: flow.name,
+            name: intentName,
+            slots: flow.variables?.map(x => ({ name: x.name, entity: x.type })) ?? [], // TODO: sub-type (type of list entity and type of pattern is not included in variables yet)
```

Comment:
does that mean that we define all variables as slots ?  e.g a variable of type number set by an action for instance ?

Reply:
yes, although if there's no training data tagged with it, then the probability of it being extracted is (near) zero